syntax = "proto3";
package asi;

service DatasetLineageService {
  rpc AddEntry(AddEntryRequest) returns (AddEntryReply);
  rpc GetEntries(GetEntriesRequest) returns (GetEntriesReply);
}

message AddEntryRequest {
  string note = 1;
  repeated string inputs = 2;
  repeated string outputs = 3;
}

message AddEntryReply {
  bool ok = 1;
}

message OutputEntry {
  string path = 1;
  string hash = 2;
  string watermark_id = 3;
}

message Entry {
  string note = 1;
  repeated string inputs = 2;
  repeated OutputEntry outputs = 3;
}

message GetEntriesRequest {}

message GetEntriesReply {
  repeated Entry entries = 1;
}
